src/ophyd_async/epics/_backend/_aioca.py

```diff
@@ -35,7 +35,58 @@
     dbr.DBR_CHAR: "string",
     dbr.DBR_LONG: "integer",
     dbr.DBR_DOUBLE: "number",
+    dbr.DBR_ENUM: "string",
 }
+_num_dbr = {dbr.DBR_SHORT, dbr.DBR_CHAR, dbr.DBR_LONG, dbr.DBR_FLOAT, dbr.DBR_DOUBLE}
+_floating_dbr = {dbr.DBR_FLOAT, dbr.DBR_DOUBLE}
+
+
+def _data_key_from_augmented_value(source: str, value: AugmentedValue) -> DataKey:
+    """Use the return value of get with FORMAT_CTRL to construct a DataKey
+    describing the signal. See docstring of AugmentedValue for expected
+    value fields by DBR type.
+
+    Args:
+        value (AugmentedValue): Description of the the return type of a DB record
+
+    Returns:
+        DataKey: A rich DataKey describing the DB record
+    """
+
+    # Always present keys
+    if not value.ok:
+        raise Exception(f"{source} error: {value.errorcode}: {str(value)}")
```

Comment:
What's the correct Exception to throw here?/ at all? Previously were not checking this.
Docs of AugmentedValue says that str(value) contains the error message- does this mean after calling signal.get()?

Reply:
We should never get here, as if `value.ok == False` then the `caget` should have raised. 

```suggestion
    assert value.ok, f"Error reading {source}: {value}"
```